src/communication/request_sync_client_server.hpp

```diff
@@ -0,0 +1,158 @@
+#ifndef REQUEST_SYNC_CLIENT_SERVER_HPP
+#define REQUEST_SYNC_CLIENT_SERVER_HPP
+
+#include "request_client_server.hpp"
+#include "request_sync_send_queue.hpp"
+
+//
+// declarations
+//
+namespace cubcomm
+{
+  /* wrapper/helper encapsulating instances of request_client_server, request_sync_send_queue
+   * and request_queue_autosend working together
+   */
+  template <typename T_OUTGOING_MSG_ID, typename T_INCOMING_MSG_ID, typename T_PAYLOAD>
+  class request_sync_client_server
+  {
+    public:
+      using request_client_server_t = cubcomm::request_client_server<T_OUTGOING_MSG_ID, T_INCOMING_MSG_ID>;
+      using incoming_request_handler_t = typename request_client_server_t::server_request_handler;
+
+    public:
+      request_sync_client_server () = default;
+      ~request_sync_client_server ();
+      request_sync_client_server (const request_sync_client_server &) = delete;
+      request_sync_client_server (request_sync_client_server &&) = delete;
+
+      request_sync_client_server &operator = (const request_sync_client_server &) = delete;
+      request_sync_client_server &operator = (request_sync_client_server &&) = delete;
+
+    public:
+      /* initialization functions should be called in mandatory order:
```

Comment:
I don't particularly like this class interface.
It resulted as such from the usecases in `page_server` and `active_tran_server`.
If there is a better option, I'm open to it.

Reply:
Done